tests/integration/test-authenticate.cpp

```diff
@@ -36,37 +36,38 @@ TEST(electronic_id_test, authenticate)
 
     EXPECT_TRUE(cardInfo);
 
-    std::cout << "Selected card: " << cardInfo->eid().name() << std::endl;
+    std::cout << "Selected card: " << cardInfo->eid().name() << '\n';
 
     byte_vector cert = cardInfo->eid().getCertificate(CertificateType::AUTHENTICATION);
 
     std::cout << "Does the reader have a PIN-pad? "
-              << (cardInfo->eid().smartcard().readerHasPinPad() ? "yes" : "no") << std::endl;
-
-    if (cardInfo->eid().authSignatureAlgorithm() != JsonWebSignatureAlgorithm::ES384
-        && cardInfo->eid().authSignatureAlgorithm() != JsonWebSignatureAlgorithm::RS256
-        && cardInfo->eid().authSignatureAlgorithm() != JsonWebSignatureAlgorithm::PS256) {
+              << (cardInfo->eid().smartcard().readerHasPinPad() ? "yes" : "no") << '\n';
+
+    switch (cardInfo->eid().authSignatureAlgorithm()) {
```

Comment:
The choice between switch and if-else should be made by the specific context and complexity of the conditions rather than a general preference. I would say that the inherent fall-through to break and additional default to exception makes the code harder to read compared with the explicit readability of if.

Reply:
I disagree. The verbosity and easy to make copy paste and logic errors with if statement is harder to read.